src/linker/Linker.Steps/UnreachableBlocksOptimizer.cs

```diff
@@ -287,186 +217,313 @@ void ProcessStack ()
-				constantResultInstruction = methodValue;
+			if (callee.HasUnknownArguments) {
+				Dictionary<MethodDefinition, Instruction?> cache = ignoreSideEffects ? cache_unknown_parameters : cache_unknown_parameters_sef;
+				if (!cache.TryGetValue (method, out result)) {
+					result = AnalyzeMethodForConstantResult (callee, callStack, ignoreSideEffects);
+					cache.Add (method, result);
+				}
+
+				return result;
 			}
 
-			return true;
+			return AnalyzeMethodForConstantResult (callee, callStack, ignoreSideEffects);
 		}
 
-		bool TryInlineBodyDependencies (ref BodyReducer reducer, bool treatUnprocessedDependenciesAsNonConst, out bool changed)
+		Instruction? GetSizeOfResult (TypeReference type)
 		{
-			changed = false;
-			bool hasUnprocessedDependencies = false;
-			var instructions = reducer.Body.Instructions;
-			Instruction? targetResult;
+			MethodDefinition? sizeOfImpl = null;
 
-			for (int i = 0; i < instructions.Count; ++i) {
-				var instr = instructions[i];
-				switch (instr.OpCode.Code) {
+			//
+			// sizeof (IntPtr) and sizeof (UIntPtr) are just aliases for IntPtr.Size and UIntPtr.Size
+			// which are simple static properties commonly overwritten. Instead of forcing C# code style
+			// we handle both via static get_Size method
+			//
+			if (type.MetadataType == MetadataType.UIntPtr) {
+				sizeOfImpl = (UIntPtrSize ??= FindSizeMethod (_context.TryResolve (type)));
+			} else if (type.MetadataType == MetadataType.IntPtr) {
+				sizeOfImpl = (IntPtrSize ??= FindSizeMethod (_context.TryResolve (type)));
+			}
 
-				case Code.Call:
-				case Code.Callvirt:
-					var md = _context.TryResolve ((MethodReference) instr.Operand);
-					if (md == null)
-						break;
+			if (sizeOfImpl == null)
+				return null;
 
-					if (md.CallingConvention == MethodCallingConvention.VarArg)
-						break;
+			return TryGetMethodCallResult (new CalleePayload (sizeOfImpl, Array.Empty<Instruction> ()), GetRecursionGuard (), ignoreSideEffects: true);
+		}
 
-					bool explicitlyAnnotated = _context.Annotations.GetAction (md) == MethodAction.ConvertToStub;
+		static Instruction? EvaluateIntrinsicCall (MethodReference method, Instruction[] arguments)
+		{
+			//
+			// In theory any pure method could be executed easily via reflection
+			// but for now we focus only on selected list method that helps
+			// reduce with framework trimming
+			//
+			object? left, right;
+			if (method.DeclaringType.MetadataType == MetadataType.String) {
+				switch (method.Name) {
+				case "op_Equality":
+				case "op_Inequality":
+				case "Concat":
+					if (arguments.Length != 2)
+						return null;
+
+					if (!GetConstantValue (arguments[0], out left) ||
+						!GetConstantValue (arguments[1], out right))
+						return null;
+
+					if (left is string sleft && right is string sright) {
+						if (method.Name.Length == 6)
+							return Instruction.Create (OpCodes.Ldstr, string.Concat (sleft, sright));
```

Comment:
This can create a very long string if called recursively.

Just curious - what are examples of CoreLib methods where this helps with trimming?

Reply:
It does not matter if it creates long string as this logic is used for result computation only (value is not inlined). I found it for example here https://cs.github.com/dotnet/runtime/blob/f85ea976f81945ea18cd5dc71959cccecdc93cd2/src/libraries/System.Private.CoreLib/src/System/TimeZoneInfo.cs?q=GetUtcFullDisplayName#L2111